### ISSUE TITLE
Visual style enhancement suggestion

### ISSUE THREAD

**Issue by shinenelson** [npmhub/npmhub#139]
After the [design refresh](https://github.blog/changelog/2020-06-23-design-updates-to-repositories-and-github-ui/) to the repository homepage, npmhub looks kind of out of style in its current form. I feel it needs a visual refresh to match with GitHub's new style. My suggestion is to match GitHub's [Dependency Graph](https://docs.github.com/en/github/visualizing-repository-data-with-graphs/about-the-dependency-graph) style.

Disclaimer : I am not a UX designer. I might be wrong with these suggestions too.

### Width

The current full-width table made sense in the old GitHub layout. With the new sidebar, the full-width table looks out of place with empty space on its top right

![image](https://user-images.githubusercontent.com/4771718/131502956-27b1b0d2-9c65-4380-9b12-57b7fd4caf0c.png)

## GitHub's Dependency Graph Style

I really like GitHub's dependency graph style and would love it if npmhub would mimic that style letting users believe that GitHub's dependency graph moved from `/network/dependents` to the repository home page.

### Table Title

The table title can be made to look more similar to the dependency graph

![image](https://user-images.githubusercontent.com/4771718/131503564-63997acd-6b66-47a1-9615-9ad86e4cf8c7.png)

Though I have one personal request to move the count to the beginning making it : 
```
<number> dependencies defined in <a href="/blob/<default-branch>/package.json">package.json</a>
```

### Dependency Repository Resolution

Instead of simply linking to the value defined in `package.json`, we could expand on the [`GitHubURL`](https://github.com/sindresorhus/refined-github/blob/main/source/github-helpers/github-url.ts) and convert the links to include the user / organization name along with the repository name.

![image](https://user-images.githubusercontent.com/4771718/131504972-3824a68d-8c28-4b74-ab8b-20e9e88ee413.png) becomes ![image](https://user-images.githubusercontent.com/4771718/131505031-6ba41246-959e-44c1-a577-5d0cf7f919a5.png)

GitHub's dependency graph probably has a complex mechanism that determines whether the name of the repository is the same as the definition in `package.json` and then determines whether to show the definition in the list or not.

![image](https://user-images.githubusercontent.com/4771718/131505743-2800c248-f6ea-4a31-9905-8f704bf4d789.png)

We could add the description of the package as a sub-text under the name of the repository.

### Version numbers

This is probably the easiest to implement as the information is already in the `package.json`. We need to parse and display it.

![image](https://user-images.githubusercontent.com/4771718/131506282-6510e07b-7163-49cf-91c4-e43a6d926a1f.png)

### Row Padding

Each row would require some extra `padding-top` and `padding-bottom` to make them visually similar to GitHub's dependency graph.

---

To summarize : 
- [ ] reduce width to fit in the container holding `.file-container` and `<readme-toc>`
- [ ] add `padding-top` and `padding-bottom` to rows
- [ ] change table heading to match with GitHub's Dependency Graph
- [ ] display version numbers of the dependencies
- [ ] resolve repository information of dependencies

**Comment by fregante** [comment]
> * [ ]  reduce width to fit in the container holding `.file-container` and `<readme-toc>`

👍 

> * add `padding-top` and `padding-bottom` to rows

👎 they're spacious enough, we don't need 100px-tall rows



> * change table heading to match with GitHub's Dependency Graph

👍 

> * display version numbers of the dependencies

🤷‍♂️ https://github.com/npmhub/npmhub/issues/56

I was thinking about it today, it would have to be off by default and at most it would be an option (and we don't have an options page yet)



> include the user / organization name along with the repository name

It might be a little long especially for already-namespaces packages like `@sindresorhus/is` but ok



> I really like GitHub's dependency graph style

I'm interested in matching it. Their UI is much more complex and it would be easier to just fetch that and show it on the page without keeping it npm-specific. So it's not `npmhub`



> has a complex mechanism

Exactly.

---

It's best to open separate issues for the features I 👍’d. Better yet if they're separate PRs because they're unlikely to be implemented by me 😃

**Comment by shinenelson** [comment]
> it would have to be off by default

I have come to believe in https://github.com/sindresorhus/refined-github/issues/2960#issuecomment-606765382

> "Disabled by default" means most people don't need it, so it doesn't belong

---

> Better yet if they're separate PRs

I will try them out when I have the time. This issue was more of a "_gathering feedback about what the maintainer thinks of these ideas_". But then I found #78 from #56 and I :facepalm:.

**Comment by fregante** [comment]
No worries. The version might be useful here, it's just that nobody cared enough about that for years. 😃 

As for the default, I do try to keep the extension simple, but it could have options.
